### PR TITLE
feat: add support for OauthS2S credential creation & other changes

### PR DESCRIPTION
**Pull request by arjuncooliitr**
## Description
This PR adds changes to support the Install templates endpoint: 
- Add new method for OauthS2S credential creation 
- Support for additional fields (`templateId` & `services`) in request & response bodies in the Swagger & methods
- Readme changes

## Related Issue
https://jira.corp.adobe.com/browse/ACNA-2841

## How Has This Been Tested?
WIP. Not tested yet

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by shazron** [comment]
Same comments I had here: https://github.com/adobe/aio-lib-console/pull/81
Are the changes just a Transporter API swagger spec drop-in?

**Comment by arjuncooliitr** [comment]
Yes, they are Transporter API swagger missing spec drop-in + adding an endpoint in index.js
Not sure how "patch" works here. Saw Michael's comment on https://github.com/adobe/aio-lib-console/pull/81 stating that the patch might be needed when the endpoint is not in the transporter swagger spec. The changes above are to the already existing endpoints in the swagger.

File: src/index.js [hunk 1]
```diff
@@ -136,6 +138,14 @@ const { DEFAULT_ENV, getCliEnv } = require('@adobe/aio-lib-env')
  * @property {string} name the role name
  */
 
+/**
+ * @typedef {object} OauthS2SIntegrationDetails
+ * @property {string} name Name
+ * @property {string} description Description
+ * @property {string} [templateId] - templateId
+ * @property {object} [services] - services
```
Comment (shazron) [inline src/index.js:146]: I believe this is an Array of objects? we should define what Object it is (see SubscribeToServices schema in swagger)
`{Array<SubscribeToServices}` then define what `SubscribeToServices` object is.

**Review by shazron — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (arjuncooliitr) [reply src/index.js]: Yes good callout, [services] is an array of `SubscribeToServices`. I have just pushed changes to accommodate this.

**Review by shazron — APPROVED** [review]

**Pull request merged**